File: leetcode/singleNumber.py
class Solution:
    # @param A, a list of integer
    # @return an integer

    def singleNumber(self, A):
        # naive way
        s1 = set()
        s2 = set()
        for i in A:
            if i not in s1 and i not in s2:
                s1.add(i)
            elif i in s1:
                s1.remove(i)
                s2.add(i)
        if s1:
            return s1.pop()
        else:
            return None

File: cpp/leetcode/singleNumber.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    int singleNumber(int A[], int n) {
        int r = 0;
        for(int i=0; i < n; i++){
            r = r^A[i];
        }
        return r;
    }
};

int main(int argc, char const *argv[])
{
    Solution* s = new Solution();
    int A[] = {1,2,3,2,3};
    cout << s->singleNumber(A, 5) << endl;
    return 0;
}
